.github/workflows/canary-integration-test.yml

```diff
@@ -173,17 +173,39 @@ jobs:
           rgw_endpoint=$(kubectl get service -n rook-ceph |  awk '/rgw/ {print $3":80"}')
           toolbox=$(kubectl get pod -l app=rook-ceph-tools -n rook-ceph -o jsonpath='{.items[*].metadata.name}')
           # pass the valid rgw-endpoint of same ceph cluster
-          timeout 15 sh -c "until kubectl -n rook-ceph exec $toolbox -- python3 /etc/ceph/create-external-cluster-resources.py --rbd-data-pool-name replicapool --rgw-endpoint $rgw_endpoint; do sleep 1 && echo 'waiting for the rgw endpoint to be validated'; done"
+          timeout 15 sh -c "until kubectl -n rook-ceph exec $toolbox -- python3 /etc/ceph/create-external-cluster-resources.py --rbd-data-pool-name replicapool --rgw-endpoint $rgw_endpoint 2> output.txt; do sleep 1 && echo 'waiting for the rgw endpoint to be validated'; done"
+          if [ -s output.txt ]; then
+            echo "script failed with stderr error"
```

Comment:
Could at least some of this be factored out to a helper function to avoid duplication?

Reply:
I am not sure if the function is allowed in this workflow

Reply:
create a function in this file https://github.com/rook/rook/blob/master/tests/scripts/github-action-helper.sh and use here

Reply:
Thanks